src/main/java/shrek/Shrek.java

```diff
@@ -0,0 +1,104 @@
+package shrek;
+
+import java.util.Scanner;
+
+import instruction.Instruction;
+import parser.Parser;
+import storage.Storage;
+import task.TaskList;
+import ui.Ui;
+import util.ShrekException;
+
+/**
+ * The main Shrek application class. Initializes and runs the task management system.
+ * This class serves as the core logic for processing user commands and managing tasks.
+ */
+public class Shrek {
+    private final Storage storage;
+    private final TaskList tasks;
+    private final Ui ui;
+
+    /**
+     * Constructs a new Shrek instance with the specified file path for data storage.
+     *
+     * @param filePath the path to the data file
+     */
+    public Shrek(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        tasks = new TaskList(storage.load());
+    }
+
+    /**
+     * Processes a user input command and returns the response.
+     * This method replaces the text-based interaction for GUI usage.
+     *
+     * @param input the user input command
+     * @return the response message from executing the command
+     */
```

Comment:
Nice and useful summary sentences that follow the standard. (i.e. "Processes" instead of "Process")
For @param and @return, note that you may choose to omit them if it is obvious or self-explanatory what they are.
And if you do include them, I think the description of each @param and @return should also start with a capital letter, and remember to include the type. The same goes for your other header comments :)
```suggestion
    /**
     * @param input The user input command
     * @return String The response message from executing the command
     */
```